xmppserver/src/main/java/org/jivesoftware/database/SequenceManager.java

```diff
@@ -217,35 +224,26 @@ private void getNextBlock(int count) {
             pstmt.setInt(2, type);
             pstmt.setLong(3, currentID);
             // Check to see if the row was affected. If not, some other process
-            // already changed the original id that we read. Therefore, this
-            // round failed and we'll have to try again.
-            success = pstmt.executeUpdate() == 1;
-            if (success) {
-                this.currentID = currentID;
-                this.maxID = newID;
+            // already changed the original id that we read. This code should be
+            // called under a lock, so this is not supposed to happen. Throw a
+            // very verbose error if it does.
+            if (pstmt.executeUpdate() == 1) {
+                final Data result = new Data(currentID, newID);
+                sequenceBlocks.put(type, result);
+                return result;
+            } else {
+                throw new IllegalStateException("Failed at attempt to obtain an ID, aborting...");
```

Comment:
As this doesn't set abortTransaction to true, it's not actually aborted? Is that desired?

FWIW, I'd default the rollback flag to true, and set it to false immediately before the the `return result;`, so it only needs to changed in one place.

Reply:
Hmm, I now see that the rollback isn't required because no update was performed. So perhaps it's simply the log message ('aborting') that is misleading?

Reply:
This code used to iterate a couple of times, before failing. I've removed that, as, under lock, that iteration should not be needed. With that, the flow became a bit... wonkey. I like the strategy that you're suggestion, as that's less ambiguous here.

Reply:
Actually, on reflection, I don't think a transaction is required at all. 

There's a single (ACID compliant) SQL statement.

If a row is updated, the returned value can be used. 
If no row is updated, there is nothing to roll back.
If a SQL exception occurs, there is nothing to roll back.

So perhaps simply remove all the unnecessary complexity?